File: app/src/androidTest/java/com/leaguechampions/utils/ElapsedTimeIdlingResource.java
package com.leaguechampions.utils;

import android.support.test.espresso.IdlingResource;

import com.leaguechampions.core.Const;

public class ElapsedTimeIdlingResource implements IdlingResource {

    private final long startTime;
    private final long waitingTime;
    private ResourceCallback resourceCallback;

    public ElapsedTimeIdlingResource() {
        this.startTime = System.currentTimeMillis();
        this.waitingTime = Const.BEHAVIOR_DELAY_MILLIS;
    }

    public ElapsedTimeIdlingResource(long waitingTime) {
        this.startTime = System.currentTimeMillis();
        this.waitingTime = waitingTime;
    }

    @Override
    public String getName() {
        return "ElapsedTimeIdlingResource: " + waitingTime + startTime;
    }

    @Override
    public boolean isIdleNow() {
        long elapsed = System.currentTimeMillis() - startTime;
        boolean idle = (elapsed >= waitingTime);
        if (idle) {
            resourceCallback.onTransitionToIdle();
        }
        return idle;
    }

    @Override
    public void registerIdleTransitionCallback(ResourceCallback resourceCallback) {
        this.resourceCallback = resourceCallback;
    }
}

File: app/src/androidTest/java/com/leaguechampions/utils/ElapsedTimeIdlingResource.kt
package com.leaguechampions.utils

import androidx.test.espresso.IdlingResource

import com.leaguechampions.core.data.local.Const

class ElapsedTimeIdlingResource(private val waitingTime: Long = com.leaguechampions.core.data.local.Const.BEHAVIOR_DELAY_MILLIS) : IdlingResource {

    private val startTime: Long = System.currentTimeMillis()
    private var resourceCallback: IdlingResource.ResourceCallback? = null

    override fun registerIdleTransitionCallback(resourceCallback: IdlingResource.ResourceCallback) {
        this.resourceCallback = resourceCallback
    }

    override fun getName(): String {
        return "ElapsedTimeIdlingResource:" + waitingTime
    }

    override fun isIdleNow(): Boolean {
        val elapsed = System.currentTimeMillis() - startTime
        val idle = elapsed >= waitingTime
        if (idle) {
            resourceCallback!!.onTransitionToIdle()
        }
        return idle
    }
}
